commons/src/main/java/io/aiven/kafka/connect/common/source/task/PartitionInPathDistributionStrategy.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.source.task;
+
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.errors.ConnectException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * PartitionInPathDistributionStrategy allows source connectors to distribute tasks based off the folder structure that
+ * partition number that is defined in that structure. for example /PREFIX/partition={{partition}}/YYYY/MM/DD/mm this
+ * will split all tasks by the number of unique partitions defined in the storage path. e.g. Task distribution in
+ * Connect with 10 Partitions and 3 tasks |Task | Partition| |0|0| |1|1| |2|2| |0|3| |1|4| |2|5| |0|6| |1|7| |2|8| |0|9|
+ */
+public final class PartitionInPathDistributionStrategy implements SourceObjectDistributionStrategy {
+    public static final String PARTITION_ID_PATTERN = "\\{\\{partition}}";
+    private final static Logger LOG = LoggerFactory.getLogger(PartitionInPathDistributionStrategy.class);
+
+    private String prefix;
+    private int maxTasks;
+
+    PartitionInPathDistributionStrategy(final int maxTasks, final String expectedPathFormat) {
+        configureDistributionStrategy(maxTasks, expectedPathFormat);
+    }
+
+    @Override
+    public boolean isPartOfTask(final int taskId, final String pathToBeEvaluated) {
+        if (pathToBeEvaluated == null || !pathToBeEvaluated.startsWith(prefix)) {
+            LOG.warn("Ignoring path {}, does not contain the preconfigured prefix {} set up at startup",
+                    pathToBeEvaluated, prefix);
+            return false;
+        }
+        String modifiedPath = StringUtils.substringAfter(pathToBeEvaluated, prefix);
+        if (!modifiedPath.contains("/")) {
+            LOG.warn("Ignoring path {}, does not contain any sub folders after partitionId prefix {}",
+                    pathToBeEvaluated, prefix);
+            return false;
+        }
+        modifiedPath = StringUtils.substringBefore(modifiedPath, "/");
```

Comment:
nit: instead of reusing, if we can assign this to another var partitionId, could be meaning full